### PR TITLE
using NTuples now

### PR DESCRIPTION
**Pull request by SimonDanisch**
This is slowly becoming master material.

### CODE REVIEW

File: src/types.jl [hunk 1]
```diff
@@ -71,33 +35,39 @@ immutable Rectangle{T} <: GeometryPrimitive
...
+export Triangle
+export GLTriangle
+export GLFace
+export GLQuad
+
+typealias Cube{T}   HyperCube{3, T}
+typealias Circle{T} HyperSphere{2, T}
+typealias Sphere{T} HyperSphere{3, T}
+
+typealias AbsolutRectangle{T} HyperRectangle{2, T}
```
Comment (sjkelly) [inline src/types.jl]: "Absolute"?

File: src/types.jl [hunk 2]
```diff
@@ -1,68 +1,32 @@
...
-@gen_fixed_size_vector("Vector",    [:x,:y,:z,:w],              1:4, true)
-@gen_fixed_size_vector("Point",     [:x,:y,:z,:w],              1:4, true)
-@gen_fixed_size_vector("Normal",    [:x,:y,:z,:w],              1:4, true)
-@gen_fixed_size_vector("UV",        [:u,:v],                    2:2, true)
-@gen_fixed_size_vector("UVW",       [:u,:v,:w],                 3:3, true)
+immutable Normal{N, T} <: FixedVector{N, T}
+    _::NTuple{N, T}
+end
+immutable TextureCoordinate{N, T} <: FixedVector{N, T}
+    _::NTuple{N, T}
```
Comment (sjkelly) [inline src/types.jl]: Why don't these have named fields?

File: src/showtest.jl [hunk 3]
```diff
@@ -0,0 +1,6 @@
+abstract Mesh
```
Comment (sjkelly) [inline src/showtest.jl]: This file doesn't seem necessary.

File: src/meshtypes.jl [hunk 4]
```diff
@@ -0,0 +1,335 @@
+abstract Mesh
```
Comment (sjkelly) [inline src/meshtypes.jl]: -1 on this. Can we keep this as 'AbstractMesh' for some API compatibility with what is on meshes right now?

File: src/HyperRectangles/HyperRectangles.jl [hunk 5]
```diff
@@ -1,127 +1,89 @@
-module HyperRectangles
```
Comment (sjkelly) [inline src/HyperRectangles/HyperRectangles.jl]: If this is no longer a submodule, I think is should be moved to the top level src dir

File: src/types.jl (hunk 1 above)
Comment (SimonDanisch) [inline src/types.jl]: Absolute in the sense that you have two points which define the rectangle, in contrast to have one point plus width/height

File: src/types.jl (hunk 2 above)
Comment (SimonDanisch) [inline src/types.jl]: they're supposed to be private, so you should not access them!
that's done via getindex and such

File: src/types.jl (hunk 1 above)
Comment (KristofferC) [inline src/types.jl]: Think his point was the misspelling.

Comment (SimonDanisch) [inline src/types.jl]: ahaha, yeah! the blindness...

File: src/types.jl (hunk 2 above)
Comment (sjkelly) [inline src/types.jl]: Ah nice!

**Pull request merged**